runtime/compiler/trj9/optimizer/JProfilingValue.cpp

```diff
@@ -204,6 +206,44 @@ TR_JProfilingValue::perform()
    }
 
 void
+TR_JProfilingValue::removeRedundantProfilingValueCalls()
+   {
+   TR::TreeTop *cursor = comp()->getStartTree();
+   TR_BitVector *alreadyProfiledValues = new (comp()->trStackMemory()) TR_BitVector();
+   while (cursor)
+      {
+      TR::Node * node = cursor->getNode();
+      if (node->isProfilingCode() &&
+         node->getOpCodeValue() == TR::treetop &&
+         node->getFirstChild()->getOpCode().isCall() &&
+         (node->getFirstChild()->getSymbolReference()->getReferenceNumber() == TR_jProfile32BitValue ||
+         node->getFirstChild()->getSymbolReference()->getReferenceNumber() == TR_jProfile64BitValue) &&
+         node->getFirstChild()->getNumChildren() == 3)
+         {
+         TR::Node *value = node->getFirstChild()->getFirstChild();
+         
+         if (alreadyProfiledValues->isSet(value->getGlobalIndex()) &&
```

Comment:
@andrewcraik  Thanks for the review. Thanks for giving some background regarding to usage of `performTransformation`. I added it in if where I check the bitVector to see if value is already profiled and if performTransformation returns `true`.